OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnPowerState.cs

```diff
@@ -0,0 +1,84 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Needs power to operate.")]
+	public class GrantConditionOnPowerStateInfo : ConditionalTraitInfo
+	{
+		[FieldLoader.Require]
+		[GrantedConditionReference]
+		[Desc("Condition to grant.")]
+		public readonly string Condition = null;
+
+		public readonly PowerState ValidPowerStates = PowerState.Low | PowerState.Critical;
```

Comment:
Can we please make this `[FieldLoader.Require]` too?  Leaving this implicit makes the yaml side less clear than it should be IMO.